Frameworks/UIKit/StarboardXaml/LayerCoordinator.cpp

```diff
@@ -752,10 +756,15 @@ void LayerCoordinator::_ApplyContentCenter(FrameworkElement^ element, Rect conte
     if (contentCenter.Equals(Rect(0, 0, 1.0, 1.0))) {
         image->NineGrid = Thickness(0, 0, 0, 0);
     } else {
-        int left = static_cast<int>(contentCenter.X * contentSize.Width);
-        int top = static_cast<int>(contentCenter.Y * contentSize.Height);
-        int right = (static_cast<int>(contentSize.Width) - (left + (static_cast<int>(contentCenter.Width * contentSize.Width))));
-        int bottom = (static_cast<int>(contentSize.Height) - (top + (static_cast<int>(contentCenter.Height * contentSize.Height))));
+        // When creating the NineGrid, we need to take the scaled image dimensions into account
+        const double contentScale = _GetContentScale(element);
+        const double imageWidth = contentSize.Width * contentScale;
```

Comment:
>ontentSize.Width * contentScale [](start = 35, length = 31)

Should we save the original width and height from SetContent--this is undoing the divide by scale done in that function.